Marlin/src/HAL/HAL_LPC1768/u8g_com_HAL_LPC1768_ssd_sw_i2c.cpp  under construction

```diff
@@ -0,0 +1,283 @@
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/*  based on U8G2 code
+
+  u8x8_byte.c
+
+  Universal 8bit Graphics Library (https://github.com/olikraus/u8g2/)
+
+  Copyright (c) 2016, olikraus@gmail.com
+  All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without modification,
+  are permitted provided that the following conditions are met:
+
+  * Redistributions of source code must retain the above copyright notice, this list
+    of conditions and the following disclaimer.
+
+  * Redistributions in binary form must reproduce the above copyright notice, this
+    list of conditions and the following disclaimer in the documentation and/or other
+    materials provided with the distribution.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+  CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+  NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+*/
+
+/*
+  software i2c,
+  ignores ACK response (which is anyway not provided by some displays)
+  also does not allow reading from the device
+*/
+
+#ifdef TARGET_LPC1768
+
+  #include <lib/u8g.h>
+
+void delayMicroseconds(uint32_t us);
+//void pinMode(int16_t pin, uint8_t mode);
+//void digitalWrite(int16_t pin, uint8_t pin_status);
+
+
+  #define I2C_SLA         (0x3c*2)
+  //#define I2C_CMD_MODE  0x080
+  #define I2C_CMD_MODE    0x000
+  #define I2C_DATA_MODE   0x040
+
+//static uint8_t I2C_speed; // 3 - 400KHz, 13 - 100KHz
+//#define SPEED_400KHz 3
+//#define SPEED_100KHz 13
+
+
+//    #define U8G_I2C_OPT_FAST 16
+
+
+
+uint8_t SCL_pin_HAL_LPC1768_sw_I2C, SCL_port_HAL_LPC1768_sw_I2C, SDA_pin_HAL_LPC1768_sw_I2C, SDA_port_HAL_LPC1768_sw_I2C;
+
+#define SPI_SPEED 2  //20: 200KHz 5:750KHz 2:3-4MHz
+
+uint8_t u8g_i2c_send_byte_sw(uint8_t data) {
+{
+  for (uint8_t i = 0; i < 9; i++) {   // 1 extra bit for the ack/nak
+
+    if (val & 0x80)
+      for (uint8_t j = 0; j < SPI_SPEED; j++) {
+        LPC_GPIO(SDA_port_HAL_LPC1768_sw_I2C)->FIOSET = LPC_PIN(SDA_pin_HAL_LPC1768_sw_I2C);
```

Comment:
I needed the speed of the FASTIO macros and wanted the flexibility of the digitalWrite routine.

We really don't have a "nice" way of specifying non-standard pins for the software SPI routines.  We either force the user to modify the pins_YOUR_BOARD.h file or modify the ultralcd_impl_DOGM.h file.

There were two general methods of getting the pin info into the com routines:
1) Hardwire the pin functions to specific pin names which forces the pins_YOUR_BOARD.h file to conform to the driver if non-standard pins are used.  This would have allowed the direct use of the FASTIO macros.  The AVR device driver for the RepRap Discount Full Graphics Smart Controller uses this method.
2) Use the U8G method of passing in the pin numbers via the argument list.  That meant that the pin numbers had to be set at run time within the com routines.

I chose the U8G method because that's what 1.1.x uses for everything except the RepRap Discount Full Graphics Smart Controller.